src/PHPStanStaticTypeMapper/TypeMapper/UnionTypeMapper.php

```diff
@@ -68,10 +66,18 @@ public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
      */
     public function mapToPhpParserNode(Type $type, string $typeKind): ?Node
     {
-        // special case for nullable
-        return $this->matchTypeForUnionedTypes($type, $typeKind);
+        $phpParserUnionType = $this->matchPhpParserUnionType($type);
+        if ($phpParserUnionType instanceof PhpParserUnionType) {
+            return $this->resolveUnionTypeNode($phpParserUnionType);
+        }
+
+        return $phpParserUnionType;
     }
 
+    /**
+     * If type is nullable, and has only one other value,
+     * this creates at least "?Type" in case of PHP 7.4 and lower
```

Comment:
nullable is since php 7.1, so the comment seems need update for php 7.1+

Reply:
Updated :+1: